### PR TITLE
upload alternative nginx proxy configuration

### PR DESCRIPTION
**Pull request by leogzl**
Some users has experience to error: `404 page not found`.

1. FIX: the path redirection from /path to -> /path/
2. FIX: the proxy configuration for better performance

### CODE REVIEW

File: docs/source/faq.rst [hunk 1]
```diff
@@ -41,6 +41,26 @@ Here is a working configuration for Nginx (tested on 1.11) to serve Portainer at
 
 Replace ``ADDRESS:PORT`` with the Portainer server/container details.
 
+Alternative nginx configuration (tested on v.1.13.7)
+
+.. code-block:: nginx-1-13-7
+
+  rewrite ^/portainer$ /portainer/ redirect;
```
Comment (deviantony) [inline docs/source/faq.rst]: Can you wrap this in the `server` block?

Comment (leogzl) [reply docs/source/faq.rst]: Hi @deviantony 
I agree, wrapping it will make it nicer to read.

**Comment by deviantony** [comment]
Do you think that the example you provide in this PR will work for previous versions of Nginx (e.g. 1.11) ? If so, I think that we should replace the existing example with the new one, it will be less confusing.

File: docs/source/faq.rst [hunk 2]
```diff
@@ -41,6 +41,34 @@ Here is a working configuration for Nginx (tested on 1.11) to serve Portainer at
 
 Replace ``ADDRESS:PORT`` with the Portainer server/container details.
 
+Alternative nginx configuration (tested on v.1.13.7)
+
+.. code-block:: nginx-1-13-7
```
Comment (deviantony) [inline docs/source/faq.rst]: Please use `nginx` lexer instead of `nginx-1-13-7`

Comment (leogzl) [reply docs/source/faq.rst]: Done. :)

**Comment by leogzl** [comment]
Hi @deviantony 

I don't think it can run on lower version. IMHO, it is recommended to use the nginx 1.13.


Leo

**Comment by deviantony** [comment]
Hi Leo, thanks for the input. I believe a single example is less confusing anyway, could you replace the existing configuration block with the new one and state the recommended Nginx version (1.13) ?

**Comment by leogzl** [comment]
I don't mind. One question, what is the purpose of `/api/websocket`?
Because on my configuration I am ignoring that but all is running well for now, at least for what I need from the portainer.

**Comment by deviantony** [comment]
The configuration block for the `/api/websocket` location is required for the console feature to work correctly. Have you tried it with your configuration? Just run a container, inspect it, go to console and use the connect button to check if everything is working well.

I believe this is required for this to work:

```
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection "upgrade";
```

**Comment by leogzl** [comment]
I got your point. We have not explored to that feature yet.
I tried and /api/websocket need different config and also /api/endpoints (Returns 404).

**Comment by deviantony** [comment]
Ok, could you update your configuration then? I believe `/api/endpoints` should not be affected.

**Comment by leogzl** [comment]
Hi @deviantony 

After we update, like this, the websocket is running well (return 101, if that is the expected result).
But we will find HTTP 404 while go to the console as you told.
From the log, we found that the /api/enpoints got affected.

Let me explore more and will send a feedback in a few days.

**Comment by deviantony** [comment]
Thanks, I'll wait for your feedback !

**Comment by nevercast** [comment]
Just dropping this here for someone to find in the future. The configuration for reverse-proxy for portainer in Caddy is
```
your.domain {
  # Add the slash on the end, maybe someone knows how to do this in the proxy directive
  redir /portainer /portainer/
  proxy /portainer/ portainer-host:9000 {
    without /portainer
    # Add Forwarded-For headers
    transparent
    # Expect websocket upgrade
    websocket
  }
}
```

**Comment by itsconquest** [comment]
I will close this as we decided to remove the proxy config examples from the FAQ as they are no longer being maintained. 

If you want to discuss reverse proxies we have a dedicated channel for this on our slack server which you can access [here](https://join.slack.com/t/portainer/shared_invite/enQtNDk3ODQ5MjI2MjI4LTcwNGYxMWQ5OGViYWZkNDY2ZjY4YTMwMTgzYmU4YmNiOTU0MDcxYmJjNTIyYmQ0MTM5Y2QwNTg3NzNkMTk5MDg)

**Pull request closed without merging**